homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -109,31 +135,38 @@ async def _async_update_data(self) -> dict:
     def get_updates(self) -> dict:
         """Fetch data from Yale."""
 
-        if self.yale is None:
-            self.yale = YaleSmartAlarmClient(
-                self.entry.data[CONF_USERNAME], self.entry.data[CONF_PASSWORD]
-            )
-
-        try:
-            arm_status = self.yale.get_armed_status()
-            cycle = self.yale.get_cycle()
-            status = self.yale.get_status()
-            online = self.yale.get_online()
-
-        except AuthenticationError as error:
-            LOGGER.error("Authentication failed. Check credentials %s", error)
-            self.hass.async_create_task(
-                self.hass.config_entries.flow.async_init(
-                    DOMAIN,
-                    context={"source": SOURCE_REAUTH, "entry_id": self.entry.entry_id},
-                    data=self.entry.data,
+        if self.yale:
+            try:
+                arm_status = self.yale.get_armed_status()
+                cycle = self.yale.get_cycle()
+                status = self.yale.get_status()
+                online = self.yale.get_online()
+
+            except AuthenticationError as error:
+                LOGGER.error("Authentication failed. Check credentials %s", error)
+                self.hass.async_create_task(
+                    self.hass.config_entries.flow.async_init(
+                        DOMAIN,
+                        context={
+                            "source": SOURCE_REAUTH,
+                            "entry_id": self.entry.entry_id,
+                        },
+                        data=self.entry.data,
+                    )
                 )
-            )
-            raise UpdateFailed from error
-
-        return {
-            "arm_status": arm_status,
-            "cycle": cycle,
-            "status": status,
-            "online": online,
-        }
+                raise UpdateFailed from error
+
+            return {
+                "arm_status": arm_status,
+                "cycle": cycle,
+                "status": status,
+                "online": online,
+            }
+
+        else:
+            return {
+                "arm_status": None,
```

Comment:
This won't work. It would raise an error during the update parsing.

Reply:
Got it. Has already changed it